### PR TITLE
Implements DoU Loss function.

### PR DESCRIPTION
**Pull request by DangerDragon1337**
`Boundary Difference over Union Loss` is a boundary aware loss function working similar to `IoU Loss` but is more specific to boundary.

It implements the formula:

$$\mathcal{L}_{DoU} = \frac{U - I}{U - \alpha I}$$

Where `U` is union of the ground truth and prediction, `I` is intersection of the ground truth and prediction and $$\alpha$$ is a hyperparameter possessing the value of `0.8`.

### CODE REVIEW

**Comment by KyberNull** [comment]
@DangerDragon1337 Also remove the duplicate phase-2/3 in training module, as they're not being used. Instead, train.py and pretrain.py are being used.

**Review by KyberNull — APPROVED** [review]
LGTM!!! 🚀
